charts/opentelemetry-collector/templates/_helpers.tpl

```diff
@@ -110,3 +110,35 @@ Return if ingress is stable.
 {{- define "ingress.isStable" -}}
   {{- eq (include "ingress.apiVersion" .) "networking.k8s.io/v1" -}}
 {{- end -}}
+
+
+{{- define "opentelemetry-collector.podAnnotations" -}}
```

Comment:
I don't think we can do this. It can lead to subtle bugs. We should have two helpers instead `opentelemetry-collector.deployment-podAnnotations` and `opentelemetry-collector.daemonset-podAnnotations`. Same for `opentelemetry-collector.podLabels`

Reply:
I can split them into deployment/daemonset specific version, but would it end up looking like

```
{{- define "opentelemetry-collector.daemonset-podAnnotations" -}}
{{- if .Values.podAnnotations }}
{{- toYaml .Values.podAnnotations }}
{{- else if .Values.agentCollector.podAnnotations }}
{{- toYaml .Values.agentCollector.podAnnotations }}
{{- end }}
{{- end }}
```

The logic is almost the same as before, would the subtle bugs you're referencing still persist?  We could merge `.Values.podAnnotations` and `.Values.agentCollector.podAnnotations`, but my philosophy so far with the refactor is to let the legacy settings completely override and of the new root-level settings.

Reply:
By subtle bugs I mean that `agentCollector.podAnnotations` potentially can be applied to daemon set and vise versa